website/docs/r/vapp_vm.html.markdown

```diff
@@ -236,7 +255,9 @@ example for usage details. **Deprecates**: `network_name`, `ip`, `vapp_network_n
   relayed). It works by querying DHCP leases on Edge Gateway. In general it is quicker than waiting
   until Guest Tools report IP addresses, but is more constrained. However this is the only option if Guest
   Tools are not present on the VM.
-  
+* `os_type` - (Optional; *v2.9+*) Operating System type. Possible values can be found in [Os Types](#os-types). Required when creating empty VM.
+* `hardware_version` - (Optional; *v2.9+*) Virtual Hardware Version (e.g.`vmx-14`, `vmx-13`, `vmx-12`, etc.). Required when creating empty VM.
+* `boot_image` - (Optional; *v2.9+*) Media name to mount as boot image which can be found in any accessible catalog. Image is mounted only during VM creation. On update if value is changed to empty it will eject the mounted media. If you want to mount an image later, please use [vcd_inserted_media](/docs/providers/vcd/r/inserted_media.html).  
```

Comment:
I have a question about this:
If I have two boot images with the same name in two different catalogs, the query will return two items, and the creation will fail. How do I say that I want the image from catalog 1 to be used in this case? In other words, how do I avoid the error  that I would not have for the vApp template, which must belong to a specific catalog? It seems to me that we have introduced a behavior inconsistency.

Reply:
As we agreed, add catalog name handling together with boot image.